File: backend-go/internal/httpapi/public_handlers.go
package httpapi

import "net/http"

func (d *Deps) handlePublicNote(w http.ResponseWriter, r *http.Request) {
	token := r.PathValue("token")
	view, err := d.Notes.GetPublicNote(r.Context(), token)
	if err != nil {
		respondDomainError(w, err)
		return
	}
	respondJSON(w, http.StatusOK, toPublicNoteViewDTO(view))
}

File: backend-rust/src/httpapi/public_handlers.rs
use std::sync::Arc;

use axum::extract::{Path, State};
use axum::Json;

use super::dto::{to_public_note_view_dto, PublicNoteViewDto};
use super::respond::ApiError;
use super::server::Deps;

pub async fn handle_public_note(
    State(deps): State<Arc<Deps>>,
    Path(token): Path<String>,
) -> Result<Json<PublicNoteViewDto>, ApiError> {
    let view = deps.notes.get_public_note(&token).await?;
    Ok(Json(to_public_note_view_dto(view)))
}
